speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -54,8 +54,12 @@ class SecuredMainPage extends Component {
               username={this.props.user.name}
               onLogout={this.props.onLogout}
               role={this.props.user.role}
+              userId={this.props.childID}
+              practitionerid={this.props.user.id}
+              unreadNum= {this.props.unreadNum}
+              fetchConversation={this.props.fetchConversation}
```

Comment:
Well, now we have multiple child components invoking this callback every 5 seconds. Think about it: This will result in every 5 seconds, _two identical network calls_ being made. Also, it's not nice that some random UI component decides the timing. What you should do instead: Have the `fetchConversation` method in `App.js` and also make the timed calls in `App.js`. The result is the App.js state, and only this result gets handed down to child components.